test/winston-logstash_test.js

```diff
@@ -31,9 +31,9 @@ describe('winston-logstash transport', function() {
   }
 
   function createTestServer(port, on_data) {
-    var server = net.createServer(port, function (socket) {
+    var server = net.createServer(function (socket) {
       socket.on('end', function () { });
-      socket.on('data', on_data);
+      socket.once('data', on_data);
```

Comment:
`data` events can fire multiple times if the payload size exceeds that of the internal buffer.